### PR TITLE
Merge Dev into feat-fast-sync

### PR DESCRIPTION
**Pull request by SamuelSchlesinger**
Merge dev into feat-fast-sync

### CODE REVIEW

File: node/src/components/deploy_acceptor.rs
Comment (rafal-ch) [inline node/src/components/deploy_acceptor.rs:503]: Is this change related to removal of `NodeId`?

File: node/src/components/linear_chain_sync.rs
Comment (rafal-ch) [inline node/src/components/linear_chain_sync.rs]: This file has been removed on `feat-fast-sync`, I believe we don't want to re-introduce it.

File: node/src/components/linear_chain_sync/blocks.rs
Comment (rafal-ch) [inline node/src/components/linear_chain_sync/blocks.rs]: I think the entire `linear_chain_sync` component has been removed on the `feat-fast-sync` branch.

File: node/src/components/linear_chain_sync/deploys.rs
Comment (rafal-ch) [inline node/src/components/linear_chain_sync/deploys.rs]: I think the entire `linear_chain_sync` component has been removed on the `feat-fast-sync` branch.

File: node/src/components/linear_chain_sync/event.rs
Comment (rafal-ch) [inline node/src/components/linear_chain_sync/event.rs]: I think the entire `linear_chain_sync` component has been removed on the `feat-fast-sync` branch.

File: node/src/components/linear_chain_sync/peers.rs
Comment (rafal-ch) [inline node/src/components/linear_chain_sync/peers.rs]: I think the entire `linear_chain_sync` component has been removed on the `feat-fast-sync` branch.

File: node/src/components/linear_chain_sync/traits.rs
Comment (rafal-ch) [inline node/src/components/linear_chain_sync/traits.rs]: I think the entire `linear_chain_sync` component has been removed on the `feat-fast-sync` branch.

File: node/src/tls.rs
Comment (rafal-ch) [inline node/src/tls.rs:84]: Does inlining of the `NodeId` require this struct to be `pub` now?

**Review by rafal-ch — COMMENTED** [review]
Overall looks good, however I see that some code removed on the `feat-fast-sync` branch got reintroduced. I mentioned it in the comments.

There are also some minor nits, but these are probably implemented correctly. I just pointed them out to double check.

File: node/src/components/linear_chain_sync.rs
Comment (SamuelSchlesinger) [reply node/src/components/linear_chain_sync.rs]: Yes, thanks for catching these.

File: node/src/tls.rs
Comment (SamuelSchlesinger) [reply node/src/tls.rs:84]: Doesn't require it, but the PR did that IIRC for something related to testing.

Comment (SamuelSchlesinger) [reply node/src/tls.rs:84]: It may actually be unnecessary now, but I shouldn't make that change on this merge PR.

File: node/src/components/deploy_acceptor.rs
Comment (SamuelSchlesinger) [reply node/src/components/deploy_acceptor.rs:503]: No, this was in the merge conflict. I'm not exactly sure if this is correct, but I recall it seeming like the most rational thing at the time.

**Review by rafal-ch — APPROVED** [review]
LGTM :+1: 
But due to the vast amount of changes, please consider getting at least one more approval before merging.

**Review by Fraser999 — APPROVED** [review]

**Pull request merged**
